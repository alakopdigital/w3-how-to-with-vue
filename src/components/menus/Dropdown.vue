<template>
  <div class="dropdown">
    <button>
      {{title}}
      <i v-if="icon" class="fa" :class="icon"></i>
    </button>
    <div class="content">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'dropdown',
  props: {
    title: {
      type: String,
      required: true
    },
    icon: String
  }
}
</script>

<style lang="scss">
.dropdown {
  position: relative;
  display: inline-block;

  button {
    background-color: #4CAF50;
    color: white;
    padding: 16px;
    font-size: 16px;
    border: none;
    cursor: pointer;
  }

  .content {
    display: none;
    position: absolute;
    background-color: #f9f9f9;
    min-width: 160px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
    z-index: 1;

    a {
      color: black;
      padding: 12px 16px;
      text-decoration: none;
      display: block;

      &:hover {
        background-color: #f1f1f1;
      }
    }
  }

  &:hover {
    button {
      background-color: #3e8e41;
    }
    .content {
      display: block;
    }
  }
}
</style>
