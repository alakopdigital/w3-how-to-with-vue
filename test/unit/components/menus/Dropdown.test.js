import { mount, renderer } from '#/unit/utils'
import Dropdown from '@/components/menus/Dropdown.vue'

describe('Dropdown.vue', () => {
  it('render default', () => {
    const wrap = mount(Dropdown, {
      propsData: {
        title: 'button'
      }
    })

    expect(wrap.hasClass('dropdown')).toBe(true)
    expect(wrap.find('button').text().trim()).toEqual('button')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('slots#default', () => {
    const wrap = mount(Dropdown, {
      propsData: {
        title: 'button'
      },
      slots: {
        default: `<a>Link</a>`
      }
    })

    expect(wrap.hasClass('dropdown')).toBe(true)
    expect(wrap.find('button').text().trim()).toEqual('button')
    expect(wrap.find('.content a').text()).toEqual('Link')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
