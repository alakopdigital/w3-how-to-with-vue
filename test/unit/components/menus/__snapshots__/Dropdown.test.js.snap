// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown.vue render default 1`] = `
<div class="dropdown"><button>
    button
  </button>
    <div class="content"></div>
</div>
`;

exports[`Dropdown.vue slots#default 1`] = `
<div class="dropdown"><button>
    button
  </button>
    <div class="content"><a>Link</a></div>
</div>
`;
